lib/jekyll/filters/url_filters.rb

```diff
@@ -29,6 +29,23 @@ def relative_url(input)
         ).normalize.to_s
       end
 
+      # Produces the path to the given asset relative to the assets directory at the
+      # domain root based on site.baseurl
+      #
+      # input - the asset file relative to the 'assets' directory. The string may also
+      #         include any subfolder(s) the asset may be a part of.
+      #           e.g. "main.css" or "css/main.css"
+      #
+      # Returns a path relative to the assets directory as a String.
+      def asset_url(input)
+        return if input.nil?
+        site = @context.registers[:site]
+        return ensure_leading_slash(input.to_s) if site.config["baseurl"].nil?
+        Addressable::URI.parse(
+          ensure_leading_slash(File.join(site.config["baseurl"], "assets", input.to_s))
```

Comment:
I have seen all problems using `File.join` for URLs when working with non-ASCII characters.

jekyll/jekyll-sitemap#106

I feel like there have been more issues, but cannot find them at the moment.